File: ClassLibraryCisepro/PROCESOSSQL/Conexion.vb
Imports System.Data.SqlClient
Imports ClassLibraryCisepro.ENUMS

Namespace ProcesosSql

    Public Class Conexion
        ''' <summary>
        ''' UNA COMEXION PARA CADA TIPO DE SISTEMA (cada empresa)
        ''' TODOS LOS METODOS DE GUARDAR, SELECCIONAR, BUSCAR, ESTAN LIGADOS A UN PROCEDIMIENTO EN BD Y SU TIPO DE RETRNO GENERAL ES DATATABLE</summary> 
        Private ReadOnly _connCisepro As SqlConnection
        Private ReadOnly _connSeportpac As SqlConnection
        Private ReadOnly _connAsenava As SqlConnection

        Sub New()
            _connCisepro = New SqlConnection(My.MySettings.Default.SysCiseproConnectionString)
            _connSeportpac = New SqlConnection(My.MySettings.Default.SysSeportpacConnectionString)
            _connAsenava = New SqlConnection(My.MySettings.Default.SysAsenavaConnectionString) 
        End Sub

        Public Function AbrirConexion(ByVal tip As TipoConexion) As SqlConnection
            Dim con As SqlConnection
            Try
                Select Case tip
                    Case TipoConexion.Seportpac
                        con = _connSeportpac
                    Case TipoConexion.Asenava
                        con = _connAsenava
                    Case Else ' CISEPRO
                        con = _connCisepro
                End Select
                If con.State = ConnectionState.Closed Then con.Open()
            Catch ex As Exception
                con = Nothing
                Console.WriteLine(ex.Message)
            End Try
            Return con
        End Function
         
        Public Sub CerrarConexion(ByVal tip As TipoConexion)
            Try 
                Select Case tip
                    Case TipoConexion.Seportpac
                        If Not _connSeportpac Is Nothing And _connSeportpac.State = ConnectionState.Open Then _connSeportpac.Close()
                    Case TipoConexion.Asenava
                        If Not _connAsenava Is Nothing And _connAsenava.State = ConnectionState.Open Then _connAsenava.Close()
                    Case Else ' CISEPRO
                        If Not _connCisepro Is Nothing And _connCisepro.State = ConnectionState.Open Then _connCisepro.Close()
                End Select
            Catch ex As Exception
                Console.WriteLine(ex.Message)
            End Try
        End Sub
         
    End Class
End Namespace

File: ClassLibrarySecurity/ProcesosSql/Conexion.cs
using System;
using System.Data.SqlClient;
using System.Data;
using ClassLibraryCisepro3.Enums;

namespace ClassLibraryCisepro3.ProcesosSql
{
    public class Conexion
    {
        private readonly SqlConnection _connCisepro;
        private readonly SqlConnection _connSeportpac;
        private readonly SqlConnection _connAsenava;

        public Conexion()
        {
            _connCisepro = new SqlConnection(Properties.Settings.Default.SysCiseproConnectionString);
            _connSeportpac = new SqlConnection(Properties.Settings.Default.SysSeportpacConnectionString);
            _connAsenava = new SqlConnection(Properties.Settings.Default.SysAsenavaConnectionString); 
        }

        public SqlConnection AbrirConexion(TipoConexion tip)
        {
            SqlConnection con;
            try
            {
                switch (tip)
                {
                    case TipoConexion.Seportpac:
                        con = _connSeportpac;
                        break;
                    case TipoConexion.Asenava:
                        con = _connAsenava;
                        break;
                    default: // CISEPRO
                        con = _connCisepro;  
                        break;
                }
                if (con.State == ConnectionState.Closed) con.Open();
            }
            catch (Exception ex)
            {
                con = null;
                Console.Write(ex.Message);
            }
            return con;
        }

        public void CerrarConexion(TipoConexion tip)
        {
            try
            { 
                switch (tip)
                {
                    case TipoConexion.Seportpac:
                        if (_connSeportpac != null && _connSeportpac.State == ConnectionState.Open) _connSeportpac.Close();
                        break;
                    case TipoConexion.Asenava:
                        if (_connAsenava != null && _connAsenava.State == ConnectionState.Open) _connAsenava.Close();
                        break;
                    default: // CISEPRO
                        if (_connCisepro != null && _connCisepro.State == ConnectionState.Open) _connCisepro.Close(); 
                        break;
                }
            }
            catch (Exception ex)
            {
                Console.Write(ex.Message);
            }
        }
    }
}
